### ISSUE TITLE
[FP] "Null pointer exception may occur here" in collection`s ->isNotEmpty() condition

### ISSUE THREAD

**Issue by PetrKaterinak** [kalessil/phpinspectionsea#1897]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended) 5.0.0.0 |
| Language level | PHP 8.2 |

Current behaviour
--
```php
if ($employee->salaries->isNotEmpty()) {
    //                                                                                                      | false positive here
    $employeeData['hourSalary'] = $employee->salaries->first()->getHourCost();
}
```

As salaries are not empty, it always must have ->first() item, so null pointer exception never happens
--

Environment details
---
PHP Storm  2023.1.3

Information from Help - About dialog

**Comment by jdreesen** [comment]
How should this bundle make the connection between the two calls? Not sure if this is possible (without help from the library itself: does the code have according Psalm/PHPStan annotations?).

Apart from that, I think we can only rely on the type information from PhpStorm here, so this should probably be fixed in PhpStorm.
